scripts/update-dockerfile.js

```diff
@@ -0,0 +1,139 @@
+        }
+
+        if (version === undefined) {
+            reject(new Error(`No version found for ${pkg}`));
+        }
+
+        resolve(version);
+    });
+});
+
+// Read the docker file to extract pkg dependencies
+const dockerFileLines = fs.readFileSync(DOCKERFILE_SRC).toString().split('\n');
+
+let inBlockToUpdate = false;
+
+// Map that will store all the text tokens that need to be replaced
+const replaceTokens = new Map();
+
+(async () => Promise.all(dockerFileLines.map(async (_, lineIndex) => {
+    const line = dockerFileLines[lineIndex];
+
+    // All the packages to update are continuous and start with RUN apk add command
+    inBlockToUpdate = Boolean(RegExp(/^RUN apk add --no-cache/).exec(line) || inBlockToUpdate);
+
+    if (inBlockToUpdate) {
+        const matches = line.matchAll(/([^= ]*)=([^= ]*)/g);
+
+        /*
+         * Index provided by the regex match is from the original line, but we need to consider that the replacement versions may not have
+         * the same size and adapt index if there is several dependencies on one line
+         */
+        let offset = 0;
+
+        for (const match of matches) {
+            const [full, pkg, currentVersion] = match;
+            const { index: matchIndex } = match;
+            log(`Fetching newer version for ${pkg}`);
+            const newVersion = await getPkgVersion(pkg);
+
+            if (newVersion === currentVersion) {
+                log(`Version ${currentVersion} is already the last for ${pkg}`);
+                continue;
+            }
+            log(`Updating ${pkg} from ${currentVersion} to ${newVersion}`);
+
+            const replaceToken = replaceTokens.get(lineIndex) ?? [];
+            if (replaceToken.length === 0) {
+                replaceTokens.set(lineIndex, replaceToken);
+            }
+
+            const replacement = `${pkg}=${newVersion}`;
+
+            replaceToken.push({
+                start: matchIndex + offset,
+                end: matchIndex + offset + full.length,
+                replacement,
+            });
+
+            // Adapt the offset
+            offset += replacement.length - full.length;
+        }
+
+        // If we have newline without backslash, we are not in the apk add command
+        inBlockToUpdate = line.endsWith('\\');
+    }
+})))().then(
+    () => {
+        const recomposedDockerfile = dockerFileLines.map((line, lineIndex) => {
+            const replaceToken = replaceTokens.get(lineIndex);
+            if (replaceToken) {
+                let updatedLine = line;
+                for (const { start, end, replacement } of replaceToken) {
+                    updatedLine = `${updatedLine.substring(0, start)}${replacement}${updatedLine.substring(end)}`;
+                }
+                return updatedLine;
+            }
+            return line;
+        }).join('\n');
+        fs.writeFileSync(DOCKERFILE_SRC, recomposedDockerfile);
+    },
+    () => {
```

Comment:
Please log the error in case of troubles such as `Docker not running on device` 🤦

Reply:
Right, done